components/bsx/Create/CreateCollection.vue

```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
-  MetaTransactionMixin,
-  ChainMixin,
-  AuthMixin,
-  PrefixMixin,
-  ApiUrlMixin
-) {
-  public base: BaseCollectionType = {
-    name: '',
-    file: null,
-    description: '',
-  }
-  public collectionDeposit = ''
-  protected id = '0'
-  protected attributes: Attribute[] = []
-  protected balanceNotEnough = false
-  public feesToken: Token = 'BSX'
-  @Ref('collectionForm') readonly collectionForm
-
-  public checkValidity() {
-    return this.collectionForm.checkValidity()
-  }
+const collectionDeposit = ref('')
+const balanceNotEnough = ref(false)
+const feesToken = ref('BSX' as Token)
+const collectionForm = ref()
+const accountId = ref()
+const isLoading = ref(false)
+const status = ref('')
 
-  get balanceNotEnoughMessage() {
-    if (this.balanceNotEnough) {
-      return this.$t('tooltip.notEnoughBalance')
-    }
-    return ''
-  }
+const checkValidity = () => {
+  return collectionForm.value?.checkValidity()
+}
 
-  @Watch('accountId', { immediate: true })
-  async onAccountIdChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
-      this.feesToken = await getFeesToken()
-    }
+const balanceNotEnoughMessage = computed(() => {
+  if (balanceNotEnough.value) {
+    return t('tooltip.notEnoughBalance')
   }
+  return ''
+})
 
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const classDeposit = getclassDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.collectionDeposit = (classDeposit + metadataDeposit).toString()
-    })
-  }
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
+watch(accountId, async (newVal, oldVal) => {
+  if (shouldUpdate(newVal, oldVal)) {
+    feesToken.value = await getFeesToken()
   }
+})
 
-  get depositOfToken() {
-    return getDeposit(this.feesToken, parseFloat(this.collectionDeposit))
-  }
+const created = async () => {
+  onApiConnect(apiUrl.value, (api) => {
+    const classDeposit = getclassDeposit(api)
+    const metadataDeposit = getMetadataDeposit(api)
+    collectionDeposit.value = (classDeposit + metadataDeposit).toString()
+  })
+}
 
-  protected async tryToEstimateTx(): Promise<string> {
-    const api = await ApiFactory.useApiInstance(this.apiUrl)
-    const cb = api.tx.utility.batchAll
-    const metadata = dummyIpfsCid()
-    const randomId = 0
-    const args = [createArgs(randomId, metadata)]
-    return estimate(this.accountId, cb, args)
+const balanceOfToken = computed(() => {
+  return getBalance(feesToken.value)
+})
+
+const depositOfToken = computed(() => {
+  return getDeposit(feesToken.value, parseFloat(collectionDeposit.value))
+})
+
+const tryToEstimateTx = async () => {
+  const api = await ApiFactory.useApiInstance(apiUrl.value)
+  const cb = api.tx.utility.batchAll
+  const metadata = dummyIpfsCid()
+  const randomId = 0
+  const args = [createArgs(randomId, metadata)]
+  return estimate(accountId.value, cb, args)
+}
+
+const checkBalanceBeforeTx = async () => {
```

Comment:
This is not being used either, do we still need this? Please confirm @roiLeo

Reply:
it's commented line 189, we might need this to check balance